### PR TITLE
 Add plugin helpers to backup plugin

### PR DESCRIPTION
**Pull request by cp2004**
cc @jneilliii 

  * [x] Your changes are not possible to do through a plugin and relevant
    to a large audience (ideally all users of OctoPrint)
  * [x] If your changes are large or otherwise disruptive: You have
    made sure your changes don't interfere with current development by
    talking it through with the maintainers, e.g. through a
    Brainstorming ticket
  * [x] Your PR targets OctoPrint's devel branch if it's a completely
    new feature, or maintenance if it's a bug fix or improvement of
    existing functionality for the current stable version (no PRs
    against master or anything else please)
  * [x] Your PR was opened from a custom branch on your repository
    (no PRs from your version of master, maintenance or devel please),
    e.g. dev/my_new_feature or fix/my_bugfix
  * [x] Your PR only contains relevant changes: no unrelated files,
    no dead code, ideally only one commit - rebase and squash your PR
    if necessary!
  * [x] Your changes follow the existing coding style
  * [x] If your changes include style sheets: You have modified the
    .less source files, not the .css files (those are generated with
    lessc)
  * [x] You have tested your changes (please state how!) - ideally you
    have added unit tests
  * [x] You have run the existing unit tests against your changes and
    nothing broke
  * [x] You have added yourself to the AUTHORS.md file :)

#### What does this PR do and why is it necessary?
Exports helpers from the backup plugin that can be used to create backups. Necessary because the global API key is removed in OctoPrint 2.0.0, and I felt nice so did this one for @jneilliii & [Backup Scheduler](https://plugins.octoprint.org/plugins/backupscheduler/) since it currently uses server side API calls, using the global API key.

#### How was it tested? How can it be tested by the reviewer?

Used this single file plugin:
https://gist.github.com/cp2004/9f4e02dd1e5bb56726637412c04060ac
Creates a backup a minute after startup, then deletes it a minute later.

Also tested UI backup is still functional, and that CLI backup was still functional.

#### Any background context you want to provide?
Backup Scheduler plugin.

#### What are the relevant tickets if any?
None

#### Screenshots (if appropriate)
None

#### Further notes
PR 4 of the day 🙂

### CODE REVIEW

**Comment by jneilliii** [comment]
The backup removed helper would be nice with for my GoogleDriveBackup plugin too. Thanks @cp2004, I hope this gets merged.

File: src/octoprint/plugins/backup/__init__.py [hunk 1]
```diff
@@ -454,6 +393,22 @@ def bodysize_hook(self, current_max_body_sizes, *args, **kwargs):
...
         return [("POST", r"/restore", 1024 * 1024 * 1024)]
 
+    # Exported plugin helpers
+    def create_backup_helper(self, exclude=None, backup_file=None):
+        """
+        Trigger a backup from a plugin or other internal call
+        Args:
+             exclude (list): Identifiers of data folders to exclude
+        """
+        if exclude is None or not (type(exclude) == list):
```
Comment (foosel) [inline src/octoprint/plugins/backup/__init__.py:403]: Better use `isinstance(exclude, list)` here, or we rule out anything that inherits from `list`.

**Review by foosel — CHANGES_REQUESTED** [review]
Some things to change: 

  * There's a bit of naming inconsistency with the backup file name. It's `backup_file` on the helpers, but `filename` and `backup_file` on the internal methods. I think that should be made consistent. `filename` might make the most sense here.
  * These helpers should be documented in the docs for the backup plugin.
  * See one annotation re an instance check

Apart from that I welcome this change and will happily merge it once the above has been taken care of!

**Comment by cp2004** [comment]
Ah, documentation, yeah... Don't worry, will take care of those points in the not-so-distant future. Will let you know when it's ready for re-review.

**Comment by cp2004** [comment]
I was going to try and get this done tonight, but writing the documentation will take a while since I will have to learn how it actually works first. It will get done, but maybe not before the Christmas cut off...

**Comment by cp2004** [comment]
All the points raised in the review have now been addressed. Documentation:
image
image

One thing I did encounter when writing the docs, was that there was no way to get a list of backups, to verify the one you wanted to delete existed. Not sure if it would be a good improvement to have a helper to export the list too. The names of the backups would be available either from the events, or from manually creating the backup using the `create_backup` helper.

Ready for re-review!

**Review by foosel — APPROVED** [review]

**Pull request merged**